uacp/endpoint_test.go

```diff
@@ -5,66 +5,78 @@
 package uacp
 
 import (
-	"net"
+	"context"
+	"net/url"
 	"testing"
-
-	"github.com/stretchr/testify/require"
 )
 
 func TestResolveEndpoint(t *testing.T) {
 	cases := []struct {
 		input   string
 		network string
-		addr    *net.TCPAddr
+		u       *url.URL
 		errStr  string
 	}{
 		{ // Valid, full EndpointURL
 			"opc.tcp://10.0.0.1:4840/foo/bar",
 			"tcp",
-			&net.TCPAddr{
-				IP:   net.IP([]byte{0x0a, 0x00, 0x00, 0x01}),
-				Port: 4840,
+			&url.URL{
+				Scheme: "opc.tcp",
+				Host:   "10.0.0.1:4840",
+				Path:   "/foo/bar",
 			},
 			"",
 		},
 		{ // Valid, port number omitted
 			"opc.tcp://10.0.0.1/foo/bar",
 			"tcp",
-			&net.TCPAddr{
-				IP:   net.IP([]byte{0x0a, 0x00, 0x00, 0x01}),
-				Port: 4840,
+			&url.URL{
+				Scheme: "opc.tcp",
+				Host:   "10.0.0.1:4840",
+				Path:   "/foo/bar",
 			},
 			"",
 		},
 		{ // Valid, hostname resolved
-			"opc.tcp://localhost:4840/foo/bar",
+			// note: see https://github.com/cunnie/sslip.io
+			"opc.tcp://www.1.1.1.1.sslip.io:4840/foo/bar",
 			"tcp",
-			&net.TCPAddr{
-				IP:   net.IP([]byte{0x7f, 0x00, 0x00, 0x01}),
-				Port: 4840,
+			&url.URL{
+				Scheme: "opc.tcp",
+				Host:   "1.1.1.1:4840",
+				Path:   "/foo/bar",
 			},
 			"",
 		},
 		{ // Invalid, schema is not "opc.tcp://"
 			"tcp://10.0.0.1:4840/foo/bar",
 			"",
 			nil,
-			"opcua: invalid endpoint tcp://10.0.0.1:4840/foo/bar",
+			"opcua: unsupported scheme tcp",
 		},
 		{ // Invalid, bad formatted schema
 			"opc.tcp:/10.0.0.1:4840/foo1337bar/baz",
 			"",
 			nil,
-			"opcua: could not resolve address foo1337bar:4840",
+			"lookup : no such host",
 		},
 	}
 
 	for _, c := range cases {
-		network, addr, err := ResolveEndpoint(c.input)
-		require.Equal(t, c.network, network, "network not equal")
-		require.Equal(t, c.addr.String(), addr.String(), "addr not equal")
+		var errStr string
+		network, u, err := ResolveEndpoint(context.Background(), c.input)
 		if err != nil {
-			require.ErrorContains(t, err, c.errStr)
+			errStr = err.Error()
+			if got, want := errStr, c.errStr; got != want {
```

Comment:
Could you change this to use `require` since I've removed `got, want` almost everywhere.